squidb-processor/src/com/yahoo/squidb/processor/plugins/defaults/properties/TableModelSpecFieldPlugin.java

```diff
@@ -83,21 +88,43 @@ protected boolean hasPropertyGeneratorForField(VariableElement field, DeclaredTy
 
     @Override
     protected PropertyGenerator getPropertyGenerator(VariableElement field, DeclaredTypeName fieldType) {
-        Class<? extends PropertyGenerator> generatorClass = generatorMap.get(fieldType);
+        Class<? extends BasicPropertyGenerator> generatorClass;
+        if (isIntegerPrimaryKey(field, fieldType)) {
+            // Force INTEGER PRIMARY KEY properties to be LongProperty, even if declared as e.g. int
+            generatorClass = getLongPropertyGenerator();
+        } else {
+            generatorClass = generatorMap.get(fieldType);
+        }
         try {
-            if (field.getAnnotation(PrimaryKey.class) != null &&
-                    BasicLongPropertyGenerator.class.equals(generatorClass)) {
-                return new BasicIdPropertyGenerator(modelSpec, field, utils);
+            BasicPropertyGenerator propertyGenerator = generatorClass.getConstructor(ModelSpec.class,
+                    VariableElement.class, AptUtils.class).newInstance(modelSpec, field, utils);
+            if ("rowid".equalsIgnoreCase(propertyGenerator.getColumnName())) {
```

Comment:
Not a bad idea at all, but let's file a separate issue for it. Technically SQLite columns can be named 'rowid', in which case they mask the internal rowid, but since we need it reserved for bookkeeping, this is a SquiDB-specific restriction.